packages/design-system/src/connect/components/document-toolbar/document-toolbar.tsx

```diff
@@ -1,56 +1,59 @@
 import { Icon } from "@powerhousedao/design-system";
+import {
+  exportDocument,
+  setSelectedNode,
+  setSelectedTimelineItem,
+  showRevisionHistory,
+  useDocumentTimeline,
+  useNodeParentFolderById,
+  useSelectedDocument,
+} from "@powerhousedao/reactor-browser";
+import type { PHDocument } from "document-model";
 import { useEffect, useState } from "react";
 import { twMerge } from "tailwind-merge";
-import type {
-  DocumentTimelineProps,
-  TimelineBarItem,
-  TimelineDividerItem,
-} from "../document-timeline/document-timeline.js";
 import { DocumentTimeline } from "../document-timeline/document-timeline.js";
+import { useDocumentUndoRedo } from "./utils/use-document-undo-redo.js";
 
 export type DocumentToolbarProps = {
-  title?: string;
+  document?: PHDocument;
   className?: string;
-  undo?: () => void;
-  redo?: () => void;
-  canUndo?: boolean;
-  canRedo?: boolean;
-  onExport?: () => void;
-  onClose: () => void;
-  onShowRevisionHistory?: () => void;
-  timelineItems?: Array<TimelineBarItem | TimelineDividerItem>;
+  disableRevisionHistory?: boolean;
   onSwitchboardLinkClick?: () => void;
   initialTimelineVisible?: boolean;
   timelineButtonVisible?: boolean;
-  onTimelineItemClick?: DocumentTimelineProps["onItemClick"];
+  onClose?: () => void;
 };
 
 export const DocumentToolbar: React.FC<DocumentToolbarProps> = (props) => {
   const {
-    undo,
-    canUndo,
-    redo,
-    canRedo,
-    title,
     onClose,
-    onExport,
     className,
-    onShowRevisionHistory,
+    document: _document,
     onSwitchboardLinkClick,
-    timelineItems = [],
-    onTimelineItemClick,
-    initialTimelineVisible = false,
     timelineButtonVisible = false,
+    disableRevisionHistory = false,
+    initialTimelineVisible = false,
   } = props;
 
+  const [selectedDocument] = useSelectedDocument();
+  const document = _document ?? selectedDocument;
+
+  const documentName = document?.header.name || undefined;
+  const parentFolder = useNodeParentFolderById(document?.header.id);
+  const handleClose = onClose ?? (() => setSelectedNode(parentFolder));
+  const handleExport = exportDocument;
+
+  const documentUndoRedo = useDocumentUndoRedo(document?.header.id);
+  const isUndoDisabled = !documentUndoRedo.canUndo;
```

Comment:
Before we had the `externalControlsEnabled` config to control this, we should either provide a `hideControls` prop or possibly a `controls` prop that defaults to `["undo", "redo", "export"]`.

Reply:
I've added an `enabledControls` prop with this config [484db9b](https://github.com/powerhouse-inc/powerhouse/pull/2017/commits/484db9b02383f77c56d68b80ff72597927525235)